Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM debian:12
+WORKDIR /src
+RUN apt-get update && apt-get install build-essential -y
+COPY . .
```

Comment:
What does this do?

Reply:
Let me go over all the process with the risk of being redundant
This does a two-stages build to generate a final image with the neper binaries without the additional dependencies necessary to build it.
 
It first creates a container based on debian bookworm, installs the compiler and necessary tools and copy current folder into the container image (COPY . .). Then we can build the binaries as we have everything inside the container.

In Line 8 we start the second stage, we use as base a smaller image with the only necessary dependencies inside, and we copy the binaries generated previously.

Reply:
Thanks for the detailed explanation, Antonio.